test/shopify-cli/theme/dev_server/app_extension_proxy_test.rb

```diff
@@ -117,10 +117,57 @@ def test_form_data_is_proxied_to_online_store
           })
         end
 
+        def test_pass_replace_templates_from_cookie_to_storefront
+          ShopifyCLI::DB
+            .stubs(:get)
+            .with(:shop)
+            .returns("dev-theme-server-store.myshopify.com")
+
+          ShopifyCLI::DB
+            .stubs(:get)
+            .with(:storefront_renderer_production_exchange_token)
+            .returns("TOKEN")
+
+          stub_request(:post, "https://dev-theme-server-store.myshopify.com/?_fd=0&pb=0")
+            .with(
+              body: {
+                "_method" => "GET",
+                "replace_extension_templates" => {
+                  "blocks" => {
+                    "blocks/block2.liquid" => @theme["blocks/block2.liquid"].read,
+                  },
+                },
+              },
+              headers: {
+                "Accept-Encoding" => "none",
+                "Authorization" => "Bearer TOKEN",
+                "Content-Type" => "application/x-www-form-urlencoded",
+                "Cookie" => http_cookie,
+                "Host" => "dev-theme-server-store.myshopify.com",
+                "X-Forwarded-For" => "",
+                "User-Agent" => "Shopify CLI",
+              }
+            )
+            .to_return(status: 200, body: "PROXY RESPONSE")
+
+          stub_session_id_request
+          response = request.get("/", "HTTP_COOKIE" => http_cookie)
+
+          assert_equal("PROXY RESPONSE", response.body)
+        end
+
         def test_multipart_is_proxied_to_online_store
-          skip # TODO: remove skip once replace_templates param is formed properly
+          skip # not sure how to match multipart file when multipart matching isn't supported for stubs
```

Comment:
need some help on this test, not sure how this should work?

Reply:
so i wasn't able to get to the root cause of why this test is failing, it should be exactly the same except its pointing to an app extension file instead of a theme file. I think it might be related to the content of the app extension file? when using byebug, i can see block1.liquid is empty, but for whatever reason the body is still getting populated with content.

because it's still failing, i still have it skipped